tasks/process_flakes.py

```diff
@@ -118,16 +114,22 @@ def generate_flake_dict(repo_id: int) -> FlakeDict:
     return flake_dict
 
 
-def update_passed_flakes(flake: Flake) -> None:
+def update_passed_flakes(
+    test_instance: TestInstance, flake: Flake, repo_id: int
+) -> None:
     flake.count += 1
     flake.recent_passes_count += 1
+
     if flake.recent_passes_count == FLAKE_EXPIRY_COUNT:
-        flake.end_date = dt.datetime.now(tz=dt.UTC)
+        flake.end_date = test_instance.created_at
+
     flake.save()
 
 
 def upsert_failed_flake(
```

Comment:
nit: we should try and be persistent about ordering of params across functions belonging to the same module

test_instance,flake,repo_id vs. test_instance,repo_id,flake

And Ideally the most common params occur first